### PR TITLE
 Test Only: Revert interpolateTypeVars fix

### PR DESCRIPTION
**Pull request by odersky**

### CODE REVIEW

**Comment by odersky** [comment]
test performance please

**Comment by liufengyun** [comment]
test performance please

**Comment by odersky** [comment]
Wow, impressive. So it was that change and the benchmarks were telling the truth!

File: compiler/src/dotty/tools/dotc/typer/Inferencing.scala [hunk 1]
```diff
@@ -412,7 +412,7 @@ trait Inferencing { this: Typer =>
     // `qualifying`.
 
     val ownedVars = state.ownedVars
-    if (ownedVars.size > 0) {
+    if (ownedVars.size > locked.size) {
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/typer/Inferencing.scala]: This seems to be a miracle -- how this change could have such a big impact on performance.

**Pull request closed without merging**